torchmultimodal/modules/layers/attention_pooler.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List
+
+import torch
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.multi_head_attention import (
+    MultiHeadAttentionWithCache,
+)
+
+
+class AttentionPooler(nn.Module):
+    """
+    Attention pooling layer: pools inputs to sequence length n_queries by performing
+        cross-attention with learned query embeddings. Based on the CoCa implementation
```

Comment:
Did CoCA invent this or just use it? It'd be better if we could point to the original abstract if this is meant as a general layer.

Reply:
Good point. I think it is originally from the set transformer paper, will add a link.